src/source/Crypto/Dtls_openssl.c

```diff
@@ -193,12 +245,16 @@ STATUS createSslCtx(PDtlsSessionCertificateInfo pCertificates, UINT32 certCount,
 
     CHK(pSslCtx != NULL, STATUS_SSL_CTX_CREATION_FAILED);
 
-    // Version greater than or equal to 1.0.2
-#if (OPENSSL_VERSION_NUMBER >= 0x10002000L)
+    // Version less than 1.1.0
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
     SSL_CTX_set_ecdh_auto(pSslCtx, TRUE);
-#else
+
+    // Version less than 3.0.0 and greater than 1.1.0
+#elif (OPENSSL_VERSION_NUMBER < 0x30000000L)
     CHK((ecdh = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1)) != NULL, STATUS_SSL_CTX_CREATION_FAILED);
     CHK(SSL_CTX_set_tmp_ecdh(pSslCtx, ecdh) == 1, STATUS_SSL_CTX_CREATION_FAILED);
+#else
+    DLOGI("ECDH enabled by default in 3.0.0. Nothing to do");
```

Comment:
Is this intended change in behavior?  Previously for openssl 1.0.0 we should call:

```
    CHK((ecdh = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1)) != NULL, STATUS_SSL_CTX_CREATION_FAILED);
    CHK(SSL_CTX_set_tmp_ecdh(pSslCtx, ecdh) == 1, STATUS_SSL_CTX_CREATION_FAILED);
```

With this change with openssl 1.0.0 we call:

```
SSL_CTX_set_ecdh_auto(pSslCtx, TRUE);
```

Reply:
Yes. It is intended. Post 1.1.0, I do not see the auto API. Its deprecated. 
https://www.openssl.org/docs/man1.1.1/man3/

Before we would invoke:
```
SSL_CTX_set_ecdh_auto(pSslCtx, TRUE);
 CHK((ecdh = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1)) != NULL, STATUS_SSL_CTX_CREATION_FAILED);
    CHK(SSL_CTX_set_tmp_ecdh(pSslCtx, ecdh) == 1, STATUS_SSL_CTX_CREATION_FAILED);
```

I just rearranged to make sure we dont call a deprecated API for versions >1.1.0

Reply:
Ok. I might have been wrong here. I just opened 1.0.0 on codespaces and could not find the API when I grep'ed it. `SSL_CTX_set_ecdh_auto` seems to have been introduced in 1.0.2. I will fix this.